### ISSUE TITLE
Error in running the example in README using irmin command line tool

### ISSUE THREAD

**Issue by priyas13** [mirage/irmin#557]
irmin set foo/bar "testing 123" leads to this error.
"irmin: unknown command `set', did you mean `dot' ?"

The example works fine when compiling and running it through OCaml

**Comment by samoht** [comment]
Which version of irmin are you using?

**Comment by priyas13** [comment]
I am using Irmin 1.4.0


Thanks,

Swarn

________________________________

**Comment by samoht** [comment]
yes you need to use the dev version for now -- we are preparing a new release in the coming weeks where this will be fixed properly.

**Comment by priyas13** [comment]
Ok. I will use OCaml to compile and run my codes but whenever I try to use module Irmin_mem, always get the error "Unbound module Irmin_mem".


________________________________

**Comment by samoht** [comment]
The best way to use the dev version is to clone everything into the same repository:

```
$ git clone https://github.com/mirage/irmin.git
$ cd irmin
$ git clone https://github.com/mirage/ocaml-git.git
$ make
$ dune exec -- irmin --root=/tmp/irmin set foo/bar "testing 123"
```

To use `Irmin_mem` you will need to use the `irmin-mem` library.

**Comment by samoht** [comment]
You need to install the missing opam dependencies too (e.g. `opam install checkseum`). You might also need to pin digestif and decompress the their dev version, e.g. `opam pin add digestif.dev --dev` and `opam pin add add digestif.dev --dev`.

**Comment by priyas13** [comment]
I tried all these steps but doesn't seem to help. I am attaching the screenshot of the error faced while installing checkseum. I tried installing checkseum using opam but faced this error.

________________________________

**Comment by samoht** [comment]
The screenshot doesn't appear on GitHub, what error message did you get?

 Anyway, @dinosaure is in the process of releasing the new dependencies of irmin, so things will improve shortly.

**Comment by priyas13** [comment]
Hello,


I am attaching the screenshot of the error here. During installation, the only problem is the installation of checkseum which in turn is creating the problem for compiling any program which uses checkseum library.


Could you direct me to the older version of Irmin git repository so that I can git clone and install the older version and try to write a few examples?


Thanks,

Swarn

________________________________

**Comment by samoht** [comment]
You need to pin checkseum too: `opam pin add checkseum.dev --dev`

**Comment by priyas13** [comment]
No luck in this also. I pinned the package and sync it but I am getting the same error. All the other dependencies are met except checkseum. Sorry for bothering you.

________________________________

**Comment by dinosaure** [comment]
@priyas13 strange error, I'm currently on `checkseum` so I will try to reproduce your error.

**Comment by priyas13** [comment]
Thanks for looking at this error. I have several other issues like not able to use the command line. There are several examples in the tutorial like

$ export EXAMPLE=/tmp/irmin/example

$ mkdir -p $EXAMPLE

$ irmin set -s git --root $EXAMPLE "My key" "My value"

$ irmin get -s git --root $EXAMPLE "My key"

My value

$ irmin remove -s git --root $EXAMPLE "My key"


I am not able to run it because irmin installed in my machine could not recognize the command 'set' and 'get'.

________________________________

**Comment by pascutto** [comment]
I can't reproduce your issue. 
Please check that you have the latest version of irmin, and feel free to reopen this if the problem persists.

**Comment by dinosaure** [comment]
According error reported by @priyas13 about `checkseum`, it's due to a bug on OCaml on MacOS (https://github.com/ocaml/ocaml/pull/1094) and fixed since 4.05.0.
